recipe/build.sh

```diff
@@ -1,36 +1,20 @@
 #!/bin/bash
+set -euxo pipefail
 
-BUILD_DIR=${SRC_DIR}/build
+BUILD_DIR="${SRC_DIR}/build"
+rm -rf "${BUILD_DIR}" || true
+mkdir -p "${BUILD_DIR}"
+cd "${BUILD_DIR}"
 
-mkdir ${BUILD_DIR}
-cd ${BUILD_DIR}
-
-PYTHON_INCLUDE_DIR=$(${PYTHON} -c 'import sysconfig;print("{0}".format(sysconfig.get_path("platinclude")))')
-PYTHON_LIBRARY=$(${PYTHON} -c 'import sysconfig;print("{0}/{1}".format(*map(sysconfig.get_config_var, ("LIBDIR", "LDLIBRARY"))))')
```

Comment:
There were many problems with cmake being overly aggressive with finding packages and python version not provided by conda. The CMAKE_FIND options and explicit Python options are being removed which is of some concern.

Are you aware of any change in conda/conda-forge behavior to help isolate cmake in the conda build environment to prevent it from finding packages outside the created build environment?

Reply:
CMake seems to find it correctly without assistance:

```
-- The C compiler identification is Clang 14.0.4
-- The CXX compiler identification is Clang 14.0.4
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: $BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- HDF5 CXX compiler wrapper is unable to compile a minimal HDF5 program.
-- HDF5 C compiler wrapper is unable to compile a minimal HDF5 program.
-- Found HDF5: $PREFIX/lib/libhdf5_cpp.dylib;$PREFIX/lib/libhdf5.dylib;$PREFIX/lib/libhdf5.dylib (found version "1.12.1") found components: CXX C 
-- Found SWIG: $BUILD_PREFIX/bin/swig (found suitable version "4.0.2", minimum required is "4")  
-- Found PythonInterp: $BUILD_PREFIX/bin/python (found version "3.10.6") 
-- Found PythonLibs: $PREFIX/lib/libpython3.10.dylib (found version "3.10.6") 
-- Not using SimpleITK's virtualenv for distribution!
```

I changed it because the previous manual mode somehow made it pick the static lib (.a) when cross compiling. Also note the previous build script was using `PYTHON_LIBRARY_DIR`, but only `PYTHON_LIBRARY` was defined, so it didn't have much of an effect.

`$CMAKE_ARGS` is part of the "magic", I guess. This is the rendered command:

```bash
cmake \
    -DCMAKE_AR="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ar" \
    -DCMAKE_CXX_COMPILER_AR="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ar" \
    -DCMAKE_C_COMPILER_AR="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ar" \
    -DCMAKE_RANLIB="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ranlib" \
    -DCMAKE_CXX_COMPILER_RANLIB="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ranlib" \
    -DCMAKE_C_COMPILER_RANLIB="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ranlib" \
    -DCMAKE_LINKER="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-ld" \
    -DCMAKE_STRIP="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-strip" \
    -DCMAKE_INSTALL_NAME_TOOL="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-install_name_tool" \
    -DCMAKE_LIBTOOL="$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-libtool" \
    -DCMAKE_OSX_DEPLOYMENT_TARGET="11.0" \
    -DCMAKE_BUILD_TYPE="Release" \
    -DCMAKE_OSX_SYSROOT="/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk" \
    -DCMAKE_INSTALL_PREFIX="$PREFIX" \
    -DCMAKE_INSTALL_LIBDIR="lib" \
    -DCMAKE_PROGRAM_PATH="$BUILD_PREFIX/bin;$PREFIX/bin" \
    -DCMAKE_SYSTEM_NAME="Darwin" \
    -DCMAKE_SYSTEM_PROCESSOR="arm64" \
    -DCMAKE_SYSTEM_VERSION="20.0.0" \
    -G Ninja \
    -D CMAKE_BUILD_TYPE:STRING=Release \
    -D SimpleITK_BUILD_DISTRIBUTE:BOOL=ON \
    -D SimpleITK_BUILD_STRIP:BOOL=ON \
    -D BUILD_SHARED_LIBS:BOOL=OFF \
    -D BUILD_TESTING:BOOL=OFF \
    -D SimpleITK_PYTHON_USE_VIRTUALENV:BOOL=OFF \
    $SRC_DIR/Wrapping/Python
```

And some env vars that are relevant:

```bash
CMAKE_PREFIX_PATH="$PREFIX"
SDKROOT="/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk"
```

(as seen in the logs for the osx-arm64 builds).